Content.Shared/_Starlight/Actions/Jump/JumpSystem.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Linq;
+using System.Numerics;
+using Content.Server.Administration.Systems;
+using Content.Shared.Actions;
+using Content.Shared.Atmos.Components;
+using Content.Shared.Atmos.EntitySystems;
+using Content.Shared.Inventory;
+using Content.Shared.Maps;
+using Content.Shared.Movement.Components;
+using Content.Shared.Throwing;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared._Starlight.Actions.Jump;
+
+//idea taked from VigersRay
+public abstract class SharedJumpSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _action = default!;
+    [Dependency] private readonly ThrowingSystem _throwing = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly IMapManager _mapMan = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly ActionContainerSystem _actionContainer = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<JumpComponent, MapInitEvent>(OnStartup);
+        SubscribeLocalEvent<JumpComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<JumpComponent, JetJumpActionEvent>(OnJump);
+        SubscribeLocalEvent<JumpComponent, GetItemActionsEvent>(OnGetItemActions);
+        SubscribeLocalEvent<JumpActionEvent>(OnJump);
+    }
+
```

Comment:
This needs a check for mob movers, as right now picking up a resomi gives you their jump that you can use, and setting them back down doesnt give them their jump action back